lib/models/device.coffee

```diff
@@ -1917,6 +1917,170 @@ getDeviceModel = (deps, opts) ->
+	# });
+	###
+	exports.isTrackingApplicationRelease = (uuidOrId, callback) ->
+		exports.get(uuidOrId, $select: 'should_be_running__release')
+		.then ({ should_be_running__release }) ->
+			return not should_be_running__release
+		.asCallback(callback)
+
+	###*
+	# @summary Get the hash of the currently tracked release for a specific device
+	# @name getTrackedReleaseHash
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @fulfil {String} - The release hash of the currently tracked release
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.getTrackedReleaseHash('7cf02a6).then(function(release) {
+	# 	console.log(release);
+	# });
+	#
+	# @example
+	# resin.models.device.getTrackedReleaseHash('7cf02a6, function(release) {
+	# 	console.log(release);
+	# });
+	###
+	exports.getTrackedReleaseHash = (uuidOrId, callback) ->
+		exports.get(uuidOrId,
+			$select: 'id'
+			$expand:
+				should_be_running__release:
+					$select: 'commit'
+				belongs_to__application:
+					$select: 'commit'
+		)
+		.then ({ should_be_running__release, belongs_to__application }) ->
+			if not isEmpty(should_be_running__release)
+				return should_be_running__release[0].commit
+			belongs_to__application[0].commit
+		.asCallback(callback)
+
+	###*
+	# @summary Set a specific device to run a particular release
+	# @name setToRelease
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @description Configures the device to run a particular release
+	# and not get updated when the current application release changes.
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @param {String} fullReleaseHash - the hash of a successful release (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.setToRelease('7cf02a6, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.device.setToRelease(123, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.device.setToRelease('7cf02a6, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847', function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.setToRelease = (uuidOrId, fullReleaseHashOrId, callback) ->
+		Promise.try ->
+			if isId(uuidOrId) and isId(fullReleaseHashOrId)
+				return {
+					deviceId: uuidOrId
+					releaseId: fullReleaseHashOrId
+				}
+
+			releaseFilterProperty = if isId(fullReleaseHashOrId) then 'id' else 'commit'
+			exports.get(uuidOrId,
+				$select: 'id'
+				$expand:
+					belongs_to__application:
+						$select: 'id'
+						$expand:
+							owns__release:
+								$top: 1
+								$select: 'id'
+								$filter:
+									"#{releaseFilterProperty}": fullReleaseHashOrId
+								# if there is a successful release for a specific commit hash
+								# then it will be the last created
```

Comment:
Is that guaranteed? Of course you can't have another successful release after the first one, but can you have a failed, or still in progress release? I don't think the model would reject those.

Also, if there are no successful releases, this will return a failed one instead, which will probably do bad things later on.

Why not just filter for successful releases?

Reply:
Taking another look at this, I don't see a reason to not filter only the succeeded releases.
I went with this approach so that the user will get the API error if the release exists but isn't successful instead of ResinReleaseNotFound.
I will add the success filter, since getting a ReleaseNotFoundError isn't totally wrong, since the description explicitly mentions that it should be a successful release.
Makes sense?